llvm/lib/Target/AIE/AIEBasePipelinerLoopInfo.cpp

```diff
@@ -627,18 +659,151 @@ void DownCountLoop::startExpand() {
   Phi->getOperand(InitIdx).setReg(NewReg);
 }
 
+/// A Zero overhead loop enters the software pipeliner with
+/// a PseudoLoopEnd as single loop terminator and
+/// a LoopStart instruction somewhere (usually close to the end)
+/// in the preheader block
+///
+class ZeroOverheadLoop : public AIEBasePipelinerLoopInfo {
+  // The instruction initializing the counter.
+  MachineInstr *Init;
+  // The instruction defining the value used the to initialize
+  // the counter.
+  MachineInstr *DefTripCount;
+  MachineBasicBlock *LoopStartBlock;
+
+public:
+  ZeroOverheadLoop(MachineInstr *EndLoop, const AIEBaseInstrInfo &TII)
+      : AIEBasePipelinerLoopInfo(EndLoop, TII) {}
+  void adjustTripCount(int TripCountAdjust) override;
+  bool accept(MachineInstr *EndLoop);
+
+  std::optional<bool> createTripCountGreaterCondition(
+      int TC, MachineBasicBlock &MBB,
+      SmallVectorImpl<MachineOperand> &Cond) override;
+};
+
+bool ZeroOverheadLoop::accept(MachineInstr *EndLoop) {
```

Comment:
the bool return here was a bit lazy on my part. We could consider to lift the acceptance enum to AIEBase level